### PR TITLE
Update tests to unmockk objects to fix inconsistent test failures

### PR DESCRIPTION
**Pull request by jack-h-wang**
This PR adjusts two unit tests that mockk objects to unmockk them at the conclusion of the test in order to resolve an inconsistent failure in other unit tests that use those objects.

Test Steps:
1. Run Gradle tasks `clean` and `package`.
2. Confirm `package` is successful (unit tests are run in the process; errors were previously observed in this process on M2 Mac environments)

## Changes
- Adjust two unit tests to unmockk objects at the end.

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] Added tests?

## Linked Issues
Addresses a change made in #9233.

### CODE REVIEW

**Review by adegolier — APPROVED** [review]
LGTM, we all saw this pass the issues we were having.

**Review by jimmyfagan — APPROVED** [review]
Looks good, hopefully this helps us keep an eye out for this kind of issue in other tests too, guessing there are probably other spots where we are missing `unmockkObject` calls that happen to not negatively affect anything.

File: prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt [hunk 1]
```diff
@@ -140,5 +140,6 @@ class CustomFhirPathFunctionTest {
         assertThat(
             (result[0] as StringType).value
         ).isEqualTo(loincCode)
+        unmockkObject(LivdLookup, Metadata)
```
Comment (jalbinson) [inline prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt:143]: Tip: using @AfterEach is generally cleaner than cleaning up directly in your test

**Review by jalbinson — APPROVED** [review]

File: prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt (hunk 1 above)
Comment (adegolier) [reply prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt:143]: I didn't think it was normal to put an @AfterEach when only half your tests use a thing.

Comment (jalbinson) [reply prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt:143]: I assume it's a noop when the object isn't currently mocked. It looks like there is also the option to call `unmockkAll` in the `@AfterEach` block to really clean everything up. If some tests are using the unmocked versions of those objects then the current way is probably the right way to do it.

https://mockk.io/#object-mocks

Comment (jimmyfagan) [reply prime-router/src/test/kotlin/fhirengine/engine/CustomFhirPathFunctionTest.kt:143]: Some combination of `unmockkAll` and `clearAllMocks` seems to be used in some combination of `@BeforeEach` and `@AfterEach` in some test files, but it's all inconsistent. I hope we can discuss this next eng sync and establish a best practice for this moving forward. There may be some cases where we don't want to unmock/clear everything between tests, so just need to talk through the options, and then document what we decide so we can reference that when making these kinds of decisions and hopefully get to some consistency eventually. Either way, it seems like we're on the same page that this is fine for this kind of quick fix, so all good for now 👍

**Comment by adegolier** [comment]
So I'm not super sure why the change I made worked, but, since we discovered yesterday that a mockObject seems to just live on forever doing its thing if it's not unmocked, I can only assume that the other tests in the suite were using the mock(s), so when the unmock was added, they started failing. I put the mock in the beforeEach and unmock in the afterEach (as opposed to before/after all) just in case there's some weirdness with threading/concurrently running tests. The best part is that it seems to require the "every" sections as well because just putting the mock/unmock in the before/afters was not enough. It didn't work until I moved the everys up there as well.

**Comment by jimmyfagan** [comment]
I think Angela's explanation makes sense, @jack-h-wang , do you want to sign off on that and get this merged in? It hasn't been affecting me, but I figure the sooner the better for those of you who are seeing this fail.

**Pull request merged**